cmd/controller/app/options/options.go

```diff
@@ -249,6 +249,7 @@ func (s *ControllerOptions) AddFlags(fs *pflag.FlagSet) {
 		"The default 'renew before expiry' time for Certificates. "+
 		"Once a certificate is within this duration until expiry, a new Certificate "+
 		"will be attempted to be issued.")
+	fs.MarkDeprecated("renew-before-expiry-duration", "Deprecated. Please set the Certificate.Spec.RenewBefore field.")
```

Comment:
Deprecated, the flag. Hides it from the controller --help output and prints a warning message to stderr when it is used:

```
go run ./cmd/controller/main.go --renew-before-expiry-duration=1h >/dev/null
Flag --renew-before-expiry-duration has been deprecated, Deprecated. Please set the Certificate.Spec.RenewBefore field.

```

Reply:
can you make an issue on the website repo to mark it as deprecated there too once this merges? Thanks!

Reply:
Done. See https://github.com/cert-manager/website/pull/377